database/factory/factory.go

```diff
@@ -0,0 +1,97 @@
+// Copyright (C) 2019-2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package factory
```

Comment:
I found it most convenient to put it here. Importing whole node pkg for the database factory seemed too much.